src/index.ts

```diff
@@ -1,21 +1,23 @@
+import { useCallback } from 'react';
 import Web3Modal, { ICoreOptions } from 'web3modal';
 import { Network, Web3Provider } from '@ethersproject/providers';
 import create from 'zustand';
-import { useEffect } from 'react';
 
 type State = {
-  provider?: Web3Provider;
-  account?: Account;
-  network?: Network;
-  web3Modal?: Web3Modal;
+  provider: Web3Provider;
+  account: Account;
+  network: Network;
+  web3Modal: Web3Modal;
 };
 
-const useStore = create<State>(_set => ({}));
+const useStore = create<Partial<State>>(_set => ({
+  web3Modal: typeof window !== 'undefined' ? new Web3Modal() : undefined,
```

Comment:
It wasn't clear to me why you were doing this inside a `useEffect` (which would trigger an immediate re-render) if we can just set this up here. However, after adding to a Next.js app, I noticed that this would fail due to "window is undefined" on the server.

I left this here and added a `window` check, but that may not be desirable if we need the `Web3Modal` instance before `connect` is called.

Let me know what the intent is here and I'll make whatever changes are needed!

Reply:
The intent was to get around the Next.js / SSR issue, so if this doesn't have that problem it should be good to merge.